### PR TITLE
Enhance flash SSI DMA example with interrupt control for stable operations

### PR DESCRIPTION
**Pull request by francdoc**
**Description:**

This PR introduces interrupt handling in the flash_ssi_dma example to enhance the stability of flash operations on the Raspberry Pi Pico. 

Based on insights from Kevin Boone's article (https://kevinboone.me/picoflash.html) and flash.h notes on stable flash memory usage, the updated code now disables interrupts during flash operations. Preventing conflicts especially with USB operations that could otherwise cause the program to hang. 

**Changes made:**

Added save_and_disable_interrupts() and restore_interrupts() around the flash operation block to prevent execution interruptions during flash memory access.

### CODE REVIEW

**Review by peterharperuk — CHANGES_REQUESTED** [review]
It's better to just use "-DPICO_STDIO_UART=0 -DPICO_STDIO_USB=1" on the cmake command line. Someone will moan about the uart not working.

**Comment by peterharperuk** [comment]
The change makes sense - see comments about sdio uage

**Comment by fchiesad** [comment]
Sure, on it!

**Comment by francdoc** [comment]
Sure, on it!

**Comment by francdoc** [comment]
Hi @peterharperuk, I just fixed the stdio. Leaving it up to user interaction on the cmake command line.

File: flash/ssi_dma/flash_ssi_dma.c [hunk 1]
```diff
@@ -67,6 +67,8 @@ uint32_t *expect = (uint32_t *) XIP_NOCACHE_NOALLOC_BASE;
 
 int main() {
     stdio_init_all();
+    
+    sleep_ms(2500);
```
Comment (lurch) [inline flash/ssi_dma/flash_ssi_dma.c:71]: If you want to "wait for minicom to connect" it's much better to use `PICO_STDIO_USB_CONNECT_WAIT_TIMEOUT_MS` - see https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_stdio_usb/include/pico/stdio_usb.h#L57 and https://github.com/raspberrypi/pico-sdk/blob/master/src/rp2_common/pico_stdio_usb/CMakeLists.txt#L21

Comment (fchiesad) [reply flash/ssi_dma/flash_ssi_dma.c]: Thank you @lurch, I modified CMakeLists.txt with the next lines and it worked:
```

if (NOT DEFINED PICO_STDIO_USB_CONNECT_WAIT_TIMEOUT_MS)
    set(PICO_STDIO_USB_CONNECT_WAIT_TIMEOUT_MS 3000)
endif()
```

Making the sleep_ms(2500) redundant.

Comment (francdoc) [reply flash/ssi_dma/flash_ssi_dma.c]: Thank you @lurch, I modified CMakeLists.txt with the next lines and it worked:
```

if (NOT DEFINED PICO_STDIO_USB_CONNECT_WAIT_TIMEOUT_MS)
    set(PICO_STDIO_USB_CONNECT_WAIT_TIMEOUT_MS 3000)
endif()
```

Making the sleep_ms(2500) redundant.

**Comment by fchiesad** [comment]
Hi Peter (@peterharperuk), with @lurch we tweaked this PR a bit!

**Comment by francdoc** [comment]
Hi @peterharperuk, I believe I made the changes you requested.

**Review by peterharperuk — APPROVED** [review]

**Pull request merged**